src/plugins/intel_cpu/src/nodes/conv.cpp

```diff
@@ -1599,27 +1677,34 @@ void Convolution::appendZeroPointsArgs() {
     }
 }
 
-
+// Due to performance issue, brgconv will only be enabled by default:
+// 1, static shape(dynamic shape may change weights layout if the input shape changes and cause performance issue: 86948)
+// 2, support amx except having input zero point.
+// 3, support avx512 without legacy postops/per channel zero point when avx512
 void Convolution::initTryBrgconvFlag() {
-    // Due to performance issue, brgconv will only be enabled by default:
-    // 1, static shape(dynamic shape may change weights layout if the input shape changes and cause performance issue: 86948)
-    // 2, support amx except having input zero point.
-    // 3, support avx512 without legacy postops/per channel zero point when avx512
-    if (!isDynamicNode()) {
-        if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx)) {
-            shouldTryBrgconv = true;
-        } else if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core)) {
-            shouldTryBrgconv = true;
-            // should remove after binary postops performance issue resolved
-            // heuristics: if it's  avx512 ISA model && it doesn't have binary post ops or per channel zero point.
-            dnnl::primitive_attr attr;
-            setPostOps(attr, outputStaticShape(), false);
-            const auto& ops = attr.get_post_ops();
-            if (ops.get()->find(dnnl::impl::primitive_kind::binary) != -1) {
-                shouldTryBrgconv = false;
-            }
+    if (isDynamicNode())
+        return;
+
+    if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core_amx)) {
+        shouldTryBrgconv = true;
+    } else if (dnnl::impl::cpu::x64::mayiuse(dnnl::impl::cpu::x64::avx512_core)) {
+        shouldTryBrgconv = true;
+        // should remove after binary postops performance issue resolved
+        // heuristics: if it's  avx512 ISA model && it doesn't have binary post ops or per channel zero point.
+        dnnl::primitive_attr attr;
+        DEBUG_LOG("setPostOps, useLegacyPostOps=false");
+        setPostOps(attr, outputStaticShape(), false);
+
+        if (attrContainsPostOp(attr, dnnl::impl::primitive_kind::binary) &&
+            attrContainsAnyOfPostOps(attr, {dnnl::impl::primitive_kind::eltwise})) {
```

Comment:
Why this condition has been changed?

Reply:
The original condition was relying on output scales which do not really exist anymore.
And they are basically replaced with binary post ops.
So if we keep the original condition, we never "try" brgemm conv then.
I tried many new heuristics and this one "to try brgemm conv only if there are only binary post ops" (and basically use depthwise otherwise) was the best so far. But as was discussed it still can be improved.

Reply:
thanks for explanation. As long as we have separate activity that refines the logic - lets leave as is.